structure-macro-impl/src/lib.rs

```diff
@@ -109,23 +107,20 @@ fn build_pack_into_fn(values: &[StructValue], fn_decl_args: &Tokens, endianness:
                             wtr.write(buf)?;
                         });
                     } else {
-                        let pointer_type = Ident::from(value.type_name().as_str());
+                        let size = mem::size_of::<usize>();
```

Comment:
The size of the pointer should be determined by the target pointer width, not the host that compiles it. The CI passes because it doesn't currently test a 32bit binary on a 64bit machine.